lib/rubocop/cop/lint/duplicate_methods.rb

```diff
@@ -0,0 +1,68 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for duplicate methods in classes and modules.
+      #
+      # @example
+      #
+      #   class SomeClass
+      #     def some_method
+      #     end
+      #     def some_method
+      #     end
+      #   end
+      class DuplicateMethods < Cop
+        MSG = 'duplicate methods at lines %s detected.'
+
+        def on_class(node)
+          _klass_name, _parent, body = *node
+          return unless body
+          names = method_names(body)
+          check_duplicate_methods(names)
+        end
+
+        def on_module(node)
+          _klass_name, body = *node
+          return unless body
+          names = method_names(body)
+          check_duplicate_methods(names)
+        end
+
+        private
+
+        def method_names(body)
+          body.child_nodes.map do |method|
```

Comment:
I would call the block argument `child` rather than using the name `method` for two things.